### ISSUE TITLE
Administration: Logo titled and linked incorrectly

### ISSUE THREAD

**Issue by connorscarolyns** [cds-snc/platform-forms-client#2385]
The GC Forms logo is incorrectly titled "Administration" It should be consistent with the other pages titled "GC Forms" The link should take users to the home page href="/form-builder"
